File: lua/01.lua
local input = io.open('../data/01.txt'):read('*a')

local total = 0

for line in input:gmatch ('[^\n]+') do
  local first, last
  for num in line:gmatch ('[0-9]') do
    first = first or num
    last = num
  end
  local line_num = tonumber (first .. last)
  total = total + line_num
end

print (total)

File: typescript/src/01.ts
import { readFileSync } from 'fs';

const input = readFileSync ('../data/01.txt', 'utf-8')

let total = 0;

input.split('\n').forEach(line => {
    let first = null;
    let last = null;

    for (const char of line) {
        if (char >= '0' && char <= '9') {
            first = first || char
            last = char
        }
    }

    if (first !== null) {
        const lineNum = parseInt (first + last, 10);
        total += lineNum;
    }
})

console.log(total)
